core/src/autogluon/core/utils/utils.py

```diff
@@ -58,23 +58,37 @@ def generate_kfold(X, y=None, n_splits=5, random_state=0, stratified=False, n_re
         return [[train_index, test_index] for train_index, test_index in kf.split(X)]
 
 
-def setup_outputdir(path, warn_if_exist=True):
+def setup_outputdir(path, warn_if_exist=True, create_dir=True, path_suffix=None):
+    if path_suffix is None:
+        path_suffix = ''
+    if path_suffix and path_suffix[-1] == os.path.sep:
+        path_suffix = path_suffix[:-1]
+    if path is not None:
+        path = f'{path}{path_suffix}'
     if path is None:
         utcnow = datetime.utcnow()
         timestamp = utcnow.strftime("%Y%m%d_%H%M%S")
-        path = f"AutogluonModels/ag-{timestamp}{os.path.sep}"
+        path = f"AutogluonModels/ag-{timestamp}{path_suffix}{os.path.sep}"
         for i in range(1, 1000):
             try:
-                os.makedirs(path, exist_ok=False)
-                break
+                if create_dir:
+                    os.makedirs(path, exist_ok=False)
+                    break
+                else:
+                    if os.path.isdir(path):
+                        raise FileExistsError
+                    break
             except FileExistsError as e:
-                path = f"AutogluonModels/ag-{timestamp}-{i:03d}{os.path.sep}"
+                path = f"AutogluonModels/ag-{timestamp}-{i:03d}{path_suffix}{os.path.sep}"
         else:
             raise RuntimeError("more than 1000 jobs launched in the same second")
```

Comment:
this block can be simplified to easier-to-read form:
```python
path = f"AutogluonModels/ag-{timestamp}{path_suffix}{os.path.sep}"
if create_dir:
    os.makedirs(path, exist_ok=False)
else:
    i = 1
    while os.path.isdir(path):
        path = f"AutogluonModels/ag-{timestamp}-{i:03d}{path_suffix}{os.path.sep}"
        i += 1
        if i > 1000:        
            raise RuntimeError("more than 1000 jobs launched in the same second")
```

Reply:
This enables a race condition, which is why the current code is more complicated to avoid race conditions by being atomic.

Reply:
Also:
```
if create_dir:
    os.makedirs(path, exist_ok=False)
```
Is not accurate to what is occurring in the code